codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -277,6 +278,23 @@ public boolean visit(Http2Stream stream) {
         }
     }
 
+    @Override
+    public void headersSent(int streamId, boolean sent) {
+        Http2Stream stream = stream(streamId);
+        assert stream != null : "Invalid stream id";
+        stream.setProperty(headersSentKey, sent);
+    }
+
+    @Override
+    public boolean headersSent(int streamId) {
+        Http2Stream stream = stream(streamId);
+        if (stream == null) {
+            return false;
```

Comment:
Good point. Then the methods on the Http2Steam should be good enough and we don't have to deal with this ambiguity for now (also reviewing on my phone so code access isn't easy)